### PR TITLE
Feature: Enhanced public views

### PR DESCRIPTION
**Pull request by filiptypjeu**
- Added public views for listing all functionaries, groups and decorations
- Added links to and from the above pages from and to the corresponding member profiles
- Added namespaces to the urls and changed to using url templates wherever possible
- Made DOM ids more consistent
- Added inital values for all date input fields
- Changed date intervals on the public views to be showed as just the year(s) if the begin-end is 1.1-31.12, except for decorations
- Added grouping of functionaries and groups on the profile pages (functionary/group is showed only once, and the date column includes the a combination of all instances)
- "Add to group" form does no longer show the groups that the member already is in
- Minor enhancements...

### CODE REVIEW

**Comment by filiptypjeu** [comment]
Thinking of making every profile accessible to logged in users, but only showing the **contact information** if the user has agreed to it. Is hiding your decorations/functionaries/groups something that you are allowed to do, or do TF have the right always show them?

**Comment by tlangens** [comment]
> Thinking of making every profile accessible to logged in users, but only showing the **contact information** if the user has agreed to it. Is hiding your decorations/functionaries/groups something that you are allowed to do, or do TF have the right always show them?

I think just showing the name is already risky with no explicit consent.

**Comment by filiptypjeu** [comment]
> > Thinking of making every profile accessible to logged in users, but only showing the **contact information** if the user has agreed to it. Is hiding your decorations/functionaries/groups something that you are allowed to do, or do TF have the right always show them?
> 
> I think just showing the name is already risky with no explicit consent.

TF member names, functionaries and various group members are already accessible to (almost) all members of TF through meeting records, the member portal and the wiki. Showing the full name is of course unnecessary... Why TF needs your middle names is a separate question, but could show the name as "F. Stenbacka" if no extra permission has been given.

**Comment by tlangens** [comment]
> > > Thinking of making every profile accessible to logged in users, but only showing the **contact information** if the user has agreed to it. Is hiding your decorations/functionaries/groups something that you are allowed to do, or do TF have the right always show them?
> > 
> > 
> > I think just showing the name is already risky with no explicit consent.
> 
> TF member names, functionaries and various group members are already accessible to (almost) all members of TF through meeting records, the member portal and the wiki. Showing the full name is of course unnecessary... Why TF needs your middle names is a separate question, but could show the name as "F. Stenbacka" if no extra permission has been given.

Middle names are good to have to differentiate between people with the same first and last names (these do happen).

It's true, if it is restricted to logged in users, that data is already necessarily (legally) available to all members through meeting protocols and such. Should be fine then.

**Comment by filiptypjeu** [comment]
> > > > Thinking of making every profile accessible to logged in users, but only showing the **contact information** if the user has agreed to it. Is hiding your decorations/functionaries/groups something that you are allowed to do, or do TF have the right always show them?
> > > 
> > > 
> > > I think just showing the name is already risky with no explicit consent.
> > 
> > 
> > TF member names, functionaries and various group members are already accessible to (almost) all members of TF through meeting records, the member portal and the wiki. Showing the full name is of course unnecessary... Why TF needs your middle names is a separate question, but could show the name as "F. Stenbacka" if no extra permission has been given.
> 
> Middle names are good to have to differentiate between people with the same first and last names (these do happen).
> 
> It's true, if it is restricted to logged in users, that data is already necessarily (legally) available to all members through meeting protocols and such. Should be fine then.

Yeah, nothing should be accessible if you are not logged in.

Middle names are most likely not written in meeting records or such, so should I hide those if the user has not given consent to it?

File: teknologr/katalogen/views.py [hunk 1]
```diff
@@ -1,77 +1,142 @@
...
+
+
+@login_required
+def decoration_ownerships(request, decoration_id):
+    decoration = get_object_or_404(Decoration,  id=decoration_id)
+
+    return render(request, 'decoration_ownerships.html', {
+        **_get_base_context(request),
+        'decoration': decoration,
+        'decoration_ownerships': DecorationOwnership.objects.filter(decoration_id=decoration_id).order_by('acquired'),
```
Comment (backjonas) [inline teknologr/katalogen/views.py]: Would it be more useful if this was in descending order like functionaries and groups?

File: teknologr/members/static/js/applicant.js [hunk 2]
```diff
@@ -50,14 +50,14 @@ $(document).ready(function() {
     });
 
     add_request_listener({
-        selector: "#makemember",
+        selector: "#make-memeber-form",
```
Comment (backjonas) [inline teknologr/members/static/js/applicant.js]: typo

File: teknologr/members/static/js/decoration.js [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
 		url: element => `/api/decorations/${element.data("id")}/`,
 	});
 	// Delete the decoration type
 	add_request_listener({
-		selector: "#deleteDecoration",
+		selector: "#delete-d-button",
 		method: "DELETE",
 		url: element => `/api/decorations/${element.data("id")}/`,
-		confirmMessage: "Vill du radera denna hedersbetygelse?",
+		confirmMessage: "Vill du radera denna hedersbetygelse och alla des ägarskap?",
```
Comment (backjonas) [inline teknologr/members/static/js/decoration.js]: des -> dess

File: teknologr/katalogen/utils.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import datetime
+from django.utils.formats import date_format 
+
+
+def create_duration_string(begin, end):
+    if begin.month == 1 and begin.day == 1 and end.month == 12 and end.day == 31:
+        return f'{begin.year}' if begin.year == end.year else f'{begin.year}-{end.year}'
+    return f'{date_format(begin)} - {date_format(end)}'
```
Comment (backjonas) [inline teknologr/katalogen/utils.py:8]: Changed this function to use localized date strings

**Comment by backjonas** [comment]
> > > > > Thinking of making every profile accessible to logged in users, but only showing the **contact information** if the user has agreed to it. Is hiding your decorations/functionaries/groups something that you are allowed to do, or do TF have the right always show them?
> > > > 
> > > > 
> > > > I think just showing the name is already risky with no explicit consent.
> > > 
> > > 
> > > TF member names, functionaries and various group members are already accessible to (almost) all members of TF through meeting records, the member portal and the wiki. Showing the full name is of course unnecessary... Why TF needs your middle names is a separate question, but could show the name as "F. Stenbacka" if no extra permission has been given.
> > 
> > 
> > Middle names are good to have to differentiate between people with the same first and last names (these do happen).
> > It's true, if it is restricted to logged in users, that data is already necessarily (legally) available to all members through meeting protocols and such. Should be fine then.
> 
> Yeah, nothing should be accessible if you are not logged in.
> 
> Middle names are most likely not written in meeting records or such, so should I hide those if the user has not given consent to it?

I agree that just showing the names of all group members/functionaries/decoration owners _should_ be fine, but since personal information can be tricky I'll double check with some more people before deploying this change

File: teknologr/katalogen/utils.py (hunk 4 above)
Comment (filiptypjeu) [reply teknologr/katalogen/utils.py:8]: Thank you! I remember I wanted to change that myself but I did not figure out how to do it.

**Review by backjonas — APPROVED** [review]

**Pull request merged**